### ISSUE TITLE
[Collection] Emergency warnings APIs for other countries

### ISSUE THREAD

**Issue by nucleus-ffm** [nucleus-ffm/foss_warn#39]
In this thread I would like to collect possible other apis that could be used to add support for other countries. I don't plan to add support now, but maybe in the future. Therefore it would be good to know if there are any Apis.

**Comment by nucleus-ffm** [comment]
**Swiss:**
Website: https://www.alert.swiss/
APP in appstore: https://play.google.com/store/apps/details?id=ch.admin.babs.alertswiss
API’s: https://www.alert.swiss/content/alertswiss-internet/en/home/_jcr_content/polyalert.alertswiss_alerts.actual.json
Format: JSON
Legal in formations: https://www.alert.swiss/de/home/meta/rechtliches.html (CC BY-NC-SA 2.5 CH)

**Comment by nucleus-ffm** [comment]
> Yes, I would like to have this alert system in my Italian city. Open system with a kind of repositories. I am not a developer, do you understand what I mean?

That sounds great. Doesn't your city already have an alert system? I don't quite understand what you mean by "open system with some kind of repositories". If you want to include your city in FOSS Warn, we need some kind of API from which we can pull the data. Preferably in the same format as we have in Germany. If you are interested in how this is handled in Germany in general, you should contact the BBK: https://www.bbk.bund.de/EN If you are interested in how the data looks like, you can have a look at the sample data here: https://nina.api.bund.dev/ (under the first tab)

**Comment by nucleus-ffm** [comment]
>  I thought there was a format similar to .xml or something like that, where just giving the app a link like that would start reading feeds (if any).

Yes, exactly. We need some kind of file in a machine readable format like Json, XML, etc. The first question is if your city or all of Italy has an alarm system. Is there an official app or website and where can I find it?

**Comment by user4andre** [comment]
**Netherlands: “NL-Alert” warning system**
(for the collection)

The Dutch also have a warning system called NL-Alert.
The messages are mainly sent via cell broadcast to everyone in the respective area (no APP required) and displayed on various display boards (e.g. at bus stops).
For people in the border regions (who may not be permanently connected to the Dutch mobile network) there are APPs for Android and iOS. 

Homepage: **https://www.nl-alert.nl/** (almost only in Dutch)
The messages are not directly available on the website. I am therefore not sure whether they can be accessed via a central interface.

Warnings are divided into warning areas. The area (or several) must be actively selected. There is apparently no automatic determination of the location.

Information on the warning areas can be found at: 
**https://www.rijksoverheid.nl/onderwerpen/veiligheidsregios-en-crisisbeheersing/veiligheidsregios**

I have not found any references to an API.
However, there are contact details for “Questions about technical operation” at www.nl-alert.nl (whatever that means):
1. management monitoring center (MMC) in Driebergen 088 - 6620300 (available 7 days a week on and 24 hours a day)
2. nl-alert.mdc(s)politie.nl (available during office hours)

**Comment by user4andre** [comment]
**Czech Republic: There is a combined emergency call and warning app**
(for the collection)

It is primarily an emergency call APP. However, it can also receive warning messages. As far as I have found, the system is still being developed. So far, only a few regions are covered for warnings.

Information on: 
https://www.zachrankaapp.cz/
https://www.zachrankaapp.cz/en/emergency-alerts

**Comment by user4andre** [comment]
**possibly interesting list from Vodafone**

Vodafone has published the following list:
https://www.vodafone.de/roaming/eu-warndienste.pdf
_(Copyright by Vodafone !)_

**Comment by user4andre** [comment]
**Great Britain: GOV.UK Emergency Alerts (API)**
(for the collection)

I have found something else:
https://github.com/alphagov/emergency-alerts-api

My knowledge is not sufficient to judge whether this is what you are looking for. The title certainly looks promising.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nucleus-ffm/foss_warn/issues/39#issuecomment-2351126739

## `GBR`

@user4andre, https://github.com/alphagov/emergency-alerts-govuk/issues/345#issue-2152911723 *might* be useful too. I presume that they're referring to https://www.gov.uk/alerts/current-alerts.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/nucleus-ffm/foss_warn/issues/39#issuecomment-2558480823

@serrq, you've no objection to both being implemented, though? Each has advantages and disadvantages depending upon the situation.

**Comment by RokeJulianLockhart** [comment]
> May you add a switch to shutdown the cell towers way?

@serrq, that would probably be worth a separate request (I presume). This issue is merely a collection of countries’ APIs.

Irrespective, I advise you mark this discussion between us as off-topic.

**Comment by nucleus-ffm** [comment]
> May you add a switch to shutdown the cell towers way?

FOSSWarn itself has nothing to do with CellBroadcast. CellBroadcast is completely handled by the operating system and can also be disabled in the operating system settings. FOSSWarn is exclusively for emergency alerts published via the internet. But as a side note, CellBroadcast is not really an issue regarding privacy as it is entirely without any acknowledgment by the devices. It is as simple as every device logged in to the cell tower will receive this message. The cell tower provider has no information about whether or not the message reached the device. For more information, see  [wikipedia](https://en.wikipedia.org/wiki/Cell_Broadcast)

**Comment by nucleus-ffm** [comment]
> Also Italy now has its own national warning system.

Thanks for the hint, but it looks like there is no open API for the alerts.

**Comment by nucleus-ffm** [comment]
Starting with version 1.0.0 FOSSWarn will use the FOSS Public Alert Server as the backend. FPAS includes all feeds listed [here](https://alert-hub-sources.s3.amazonaws.com/json) (every feed that has the status "operating")  and all additional feeds listed [here](https://invent.kde.org/webapps/foss-public-alert-server/-/blob/master/foss_public_alert_server/sourceFeedHandler/custom_feeds.json). We are open to add more feeds to our custom feeds list. 

Maybe to make it more clear what we need to include a new feed in FPAS and, therefore, in the future also in FOSSWarn is: 

- the feeds have to provide alerts in the [CAP](https://en.wikipedia.org/wiki/Common_Alerting_Protocol) format. As this is an international standard, most countries use it anyway
- we have to be allowed to distribute the alerts
- the API has to be stable and reachable
- we have to verify that it is an official feed
